### PR TITLE
Use docker-compose run, always use npm install

### PR DESCRIPTION
**Pull request by blrubenstein**

### CODE REVIEW

File: ui/docker-compose.yaml [hunk 1]
```diff
@@ -6,7 +6,7 @@ services:
     working_dir: /w/ui
     volumes:
       - ..:/w:cached
-    command: npm start -- --host=0.0.0.0 ${ENV_FLAG}
+    command: sh -c 'npm start -- --host=0.0.0.0 ${ENV_FLAG}'
```
Comment (markfickett) [inline ui/docker-compose.yaml]: What does this do [differently]? (Maybe leave a comment?)

File: ui/libproject/devstart.rb [hunk 2]
```diff
@@ -56,23 +56,21 @@ def dev_up(*args)
 
   ensure_git_hooks
 
-  unless Dir.exist?("node_modules")
-    install_dependencies
-  end
+  install_dependencies
```
Comment (markfickett) [inline ui/libproject/devstart.rb]: Will this start a Docker container, do the install, and then kill that container before continuing on to the main startup? How is it for speed? I'm wondering if doing 'npm install' inside the main UI Docker container would be faster, if this requires more/extra setup.

File: ui/docker-compose.yaml (hunk 1 above)
Comment (blrubenstein) [reply ui/docker-compose.yaml]: This got around a problem with the host not opening. @dmohs has used this workaround before, and might have more details on why that was important.

I can ask him what exactly the purpose was and throw a comment in there.

File: ui/libproject/devstart.rb (hunk 2 above)
Comment (blrubenstein) [reply ui/libproject/devstart.rb]: I believe this actually does this on the host machine, but I'm not certain. I can do some more investigation there and get back to this.

**Comment by blrubenstein** [comment]
I just pushed a new version here, so its ready to re-review. One note is that everyone will need to run `docker-compose build` after this goes in.

File: ui/libproject/devstart.rb (hunk 2 above)
Comment (blrubenstein) [reply ui/libproject/devstart.rb]: The answer is that it actually does it inside the docker image, and that translates back down to the host machine. It seems pretty fast when I run it on my machine. (I didn't super notice the difference in speed.)

**Pull request merged**